### DISCUSSION TITLE
Homing Offsets

### DISCUSSION THREAD

**Discussion by andrewmarles** [grblHAL/core#165] [Ideas]
Just working with some different folks and machines and I don't think that homing offsets are currently implemented in GRBLHAL.  Is there a way to offset the MCS from the homing switch if it needs to be located in the middle of travel, but you still want to enforce soft limts etc.  An example of this is the head of an OpenPNP machine where two nozzles share a single Z motor and homing sensor:

![image](https://user-images.githubusercontent.com/6061539/181349918-f9f30136-c320-4bf9-9805-90377bc15466.png)

Is there a way to handle this today, or is this something that we need to work on implementing?

Cheers!

**Comment by terjeio** [comment]
This needs work, IIRC FluidNC supports this so perhaps check how it is done there?
Perhaps we should [add an event](https://github.com/grblHAL/core/blob/master/core_handlers.h), `grbl.on_homing_completed` so that the initial implementation could be added in a plugin?

**Comment by andrewmarles** [comment]
That event sounds like it would come in handy for this and other plugins.  Fluid looks like they do it by just forcing the mpos after homing and then they have some helper functions to figure out the min and max position to use for soft limit checks.  Not sure if the limit check functions could be done in a plug-in?

**Reply by terjeio** [reply]
I have already added the event to the latest core.

> Not sure if the limit check functions could be done in a plug-in?

As a start, why not? When verified then add them to the core.

BTW I will be "offline" for a while - unsure for how long, at least a few weeks.
